src/SprykerEco/Zed/CrefoPay/Persistence/CrefoPayRepositoryInterface.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/**
+ * MIT License
+ * For full license information, please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace SprykerEco\Zed\CrefoPay\Persistence;
+
+use Generated\Shared\Transfer\PaymentCrefoPayTransfer;
+
+interface CrefoPayRepositoryInterface
+{
+    /**
+     * @param string $reference
+     *
+     * @return \Generated\Shared\Transfer\PaymentCrefoPayTransfer
+     */
+    public function getPaymentCrefoPayByReference(string $reference): PaymentCrefoPayTransfer;
+
+    /**
+     * @param int $idSalesOrder
+     *
+     * @return \Generated\Shared\Transfer\PaymentCrefoPayTransfer
+     */
+    public function getPaymentCrefoPayByIdSalesOrder(int $idSalesOrder): PaymentCrefoPayTransfer;
+
+    /**
+     * @param string $pspReference
+     *
+     * @return \Generated\Shared\Transfer\PaymentCrefoPayTransfer
+     */
+    public function getPaymentCrefoPayByPspReference(string $pspReference): PaymentCrefoPayTransfer;
+
+    /**
+     * @param int $idSalesOrder
+     *
+     * @return \Generated\Shared\Transfer\PaymentCrefoPayOrderItemTransfer[]
```

Comment:
Core uses `XxxxCollectionTransfer` in this case in order to have strict typing here